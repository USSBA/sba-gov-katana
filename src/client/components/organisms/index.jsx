/* IMPORTS */

// miscellaenous organisms
import BusinessGuideTileCollection from "./business-guide-tile-collection/business-guide-tile-collection.jsx";
import CardCollection from "./card-collection/card-collection.jsx";
import DeveloperTester from "./developer-tester/developer-tester.jsx";
import DocumentCardCollection from "./document-card-collection/document-card-collection.jsx";
import FrontPageHero from "./front-page-hero/front-page-hero.jsx";
import FundingProgramsTileCollection from "./funding-programs-tile-collection/funding-programs-tile-collection.jsx";
import DisasterAlert from "./header-footer/disaster-alert/disaster-alert.jsx";
import NotificationBar from "./header-footer/notification-bar.jsx";
import Hero from "./hero/hero.jsx";
import RelatedDocumentCards from "./related-document-cards/related-document-cards.jsx";
import SearchBox from "./search-box/search-box.jsx";
import SectionNav from "./section-nav/section-nav.jsx";

// header-footer organisms
import Footer from "./header-footer/footer/footer.jsx";
import MobileNav from "./header-footer/mobile-nav/mobile-nav.jsx";
import Header from "./header-footer/header/header.jsx";

// homepage organisms
import Blog from "./homepage/blog/blog.jsx";
import FrontPageLady from "./homepage/front-page-lady/front-page-lady.jsx";
import HappeningNow from "./homepage/happening-now/happening-now.jsx";

// lender match organisms
// -- form
import AdditionalInfoForm from "./lender-match/form/additional-info.jsx";
import BusinessInfoForm from "./lender-match/form/business-info.jsx";
import ContactInfoForm from "./lender-match/form/contact-info.jsx";
import IndustryInfoForm from "./lender-match/form/industry-info.jsx";
import LoanInfo from "./lender-match/form/loan-info.jsx";
import ReviewSection from "./lender-match/form/review-page-helpers.jsx";
import ReviewSubmitInfoForm from "./lender-match/form/review-submit-info.jsx";
// -- landing page
import FindLendersIntro from "./lender-match/landing-page/find-lenders-intro.jsx";
import HelpfulQuestions from "./lender-match/landing-page/helpful-questions.jsx";
import {HowItWorksSection} from "./lender-match/landing-page/how-it-works.jsx";
import PreparationChecklist from "./lender-match/landing-page/preparation-checklist.jsx";
// -- success page
import ConfirmSection from "./lender-match/success-page/confirm-section.jsx";
import DynamicCounselingAndTools from "./lender-match/success-page/counseling-and-tools.jsx";
import CounselorMap from "./lender-match/success-page/counselor-map.jsx";
import EmailConfirmationInvalid from "./lender-match/success-page/email-confirmed-page.jsx";

// modal organisms
import LeaveSbaModal from "./modals/leave-sba-modal/leave-sba-modal.jsx";
import MobileSectionNavModal from "./modals/mobile-section-nav/mobile-section-nav.jsx";

/* EXPORTS */

// miscellaneous organisms
export {
	BusinessGuideTileCollection,
	CardCollection,
	DeveloperTester,
	DocumentCardCollection,
	FrontPageHero,
	FundingProgramsTileCollection,
	DisasterAlert,
	NotificationBar,
	Hero,
	RelatedDocumentCards,
	SearchBox,
	SectionNav
};

// header-footer organisms
export {
	Footer,
	MobileNav,
	Header
};

// homepage organisms
export {
	Blog,
	FrontPageLady,
	HappeningNow
};

// lender match organisms
export {
	AdditionalInfoForm,
	BusinessInfoForm,
	ContactInfoForm,
	IndustryInfoForm,
	LoanInfo,
	ReviewSection,
	ReviewSubmitInfoForm,
	FindLendersIntro,
	HelpfulQuestions,
	HowItWorksSection,
	PreparationChecklist,
	ConfirmSection,
	DynamicCounselingAndTools,
	CounselorMap,
	EmailConfirmationInvalid
};

// modal organisms
export {
	LeaveSbaModal,
	MobileSectionNavModal
};
