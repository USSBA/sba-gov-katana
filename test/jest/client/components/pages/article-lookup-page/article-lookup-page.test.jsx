import React from 'react'
import { shallow } from 'enzyme'
import ArticleLookupPage from 'pages/article-lookup-page/article-lookup-page'
import { PagingLookup } from 'organisms'

describe('ArticleLookupPage', () => {
  test('should render a page lookup', () => {
    // an article lookup page is just renders a paging lookup and nothing more
    const component = shallow(<ArticleLookupPage />)
    const result = component.find(PagingLookup).length
    const expected = 1
    expect(result).toEqual(expected)
  })
})
