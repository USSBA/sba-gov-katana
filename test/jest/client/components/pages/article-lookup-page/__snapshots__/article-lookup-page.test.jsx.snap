// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArticleLookupPage renders a page lookup with the correct attributes 1`] = `
<Connect(PagingLookup)
  fieldsToShowInDetails={
    Array [
      "Program",
      "Published",
      "Summary",
    ]
  }
  sortByOptions={
    Array [
      "Last Updated",
      "Title",
    ]
  }
  taxonomyFilters={
    Array [
      "articleType",
      "program",
    ]
  }
  title="Article Lookup"
  type="articles"
/>
`;
